tests/test_oauth.py

```diff
@@ -134,3 +134,37 @@ def test_oauth_user_info_no_provider(self):
         csm = CustomSecurityManager(appbuilder=appbuilder)
         user_info = csm.oauth_user_info(provider=None)
         assert user_info is None
+
+    @patch("superset_patchup.oauth.SupersetSecurityManager.clean_perms")
+    @patch("superset_patchup.oauth.SupersetSecurityManager.get_session")
+    @patch(
+        "superset_patchup.oauth.SupersetSecurityManager.create_missing_perms")
+    @patch("superset_patchup.oauth.CustomSecurityManager.set_custom_role")
+    @patch("superset_patchup.oauth.get_complex_env_var")
+    @patch(
+        "superset_patchup.oauth.SupersetSecurityManager.sync_role_definitions")
+    def test_custom_roles(
+            self,
+            mock_super,
+            mock_vars,
+            mock_set_custom_role,
+            mock_create_missing_perms,
+            mock_get_session,
+            mock_clean_perms,
+    ):
+        """
+        Test that when add custom roles is set to true, the roles specified
+        in the configs are created
+        """
+        # Sample roles
+        custom_roles = {"Test_role": "{'all_datasource_access'}"}
```

Comment:
Is this really how the roles are defined?  Looks like a weird data structure for python.